txpipe/mapping/basic_maps.py

```diff
@@ -119,41 +131,42 @@ def finalize(self, comm=None):
             v_g1 /= count_g1
             v_g2 /= count_g2
 
+            # Convert mean weight to total weight
+            weight = mean_w * count_w
+            del mean_w, count_w
+
             # Update the mask
             mask[count_g1>0] = True
             mask[count_g2>0] = True
 
-            #  Don't think we want to save these at the moment
-            del count_g1
-            del count_g2
 
+            # Repalce NaNs with the Healpix unseen sentinel value
+            # -1.63775e30
             mean_g1[np.isnan(mean_g1)] = UNSEEN
             mean_g2[np.isnan(mean_g2)] = UNSEEN
-
-
+            mean_g2[np.isnan(mean_g2)] = UNSEEN
             v_g1[np.isnan(v_g1)] = UNSEEN
             v_g2[np.isnan(v_g2)] = UNSEEN
+            weight[np.isnan(weight)] = UNSEEN
 
-            mean_g1 = mean_g1.reshape(self.pixel_scheme.shape)
-            mean_g2 = mean_g2.reshape(self.pixel_scheme.shape)
-
-            v_g1 = v_g1.reshape(self.pixel_scheme.shape)
-            v_g2 = v_g2.reshape(self.pixel_scheme.shape)
-
-            g1[b] = mean_g1.flatten()
-            g2[b] = mean_g2.flatten()
+            # I really don't recall why I'm changing the shape
+            # back and forth here.  Maybe I had a reason?
+            # TODO: Remove the reshape/flatten and see what happens.
+            g1[b] = mean_g1.reshape(self.pixel_scheme.shape).flatten()
```

Comment:
This looks harmless but yeah, I don't see the reason for the reshaping, I would just try to remove the reshapes before flattening.